nilearn/_utils/estimator_checks.py

```diff
@@ -1499,16 +1557,84 @@ def check_multi_masker_transformer_sample_mask(estimator):
     for ts, n_scrub in zip(signals_list, [n_scrub1, n_scrub2]):
         assert ts.shape[0] == length - n_scrub
 
+    # should also work with a single 4D image (has no __iter__ )
+    signals_list = estimator.fit_transform(
+        _img_4d_rand_eye_medium(),
+        sample_mask=[sample_mask1],
+    )
+
+    assert signals_list.shape[0] == length - n_scrub1
+
     with pytest.raises(
         ValueError,
         match="number of sample_mask .* unequal to number of images",
     ):
         estimator.fit_transform(
             [_img_4d_rand_eye_medium(), _img_4d_rand_eye_medium()],
-            sample_mask=sample_mask1,
+            sample_mask=[sample_mask1],
+        )
+
+    with pytest.raises(
+        TypeError, match="'sample_mask' must be a None or a list."
+    ):
+        estimator.fit_transform(
+            [_img_4d_rand_eye_medium(), _img_4d_rand_eye_medium()],
+            sample_mask=1,
         )
 
 
+def check_multi_masker_transformer_high_variance_confounds(estimator):
+    """Check high_variance_confounds use in multi maskers with 5D data.
+
+    Make sure that using high_variance_confounds returns different result.
+
+    Ensure that high_variance_confounds can be used with regular confounds,
+    and that results are different than when just using the confounds alone.
+    """
+    length = 20
+
+    data = _rng().random((*_shape_3d_default(), length))
+    input_img = Nifti1Image(data, _affine_eye())
+
+    estimator.high_variance_confounds = False
+
+    signal = estimator.fit_transform([input_img, input_img])
+
+    estimator = clone(estimator)
+    estimator.high_variance_confounds = True
+
+    signal_hvc = estimator.fit_transform([input_img, input_img])
+
+    for s1, s2 in zip(signal, signal_hvc):
+        assert_raises(AssertionError, assert_array_equal, s1, s2)
+
+    with TemporaryDirectory() as tmp_dir:
+        array = _rng().random((length, 3))
+
+        dataframe = pd.DataFrame(array)
+
+        tmp_dir = Path(tmp_dir)
+        dataframe.to_csv(tmp_dir / "confounds.csv")
+
+        for c in [array, dataframe, tmp_dir / "confounds.csv"]:
+            confounds = [c, c]
+
+            estimator = clone(estimator)
+            estimator.high_variance_confounds = False
+            signal_c = estimator.fit_transform(
+                [input_img, input_img], confounds=confounds
+            )
+
+            estimator = clone(estimator)
+            estimator.high_variance_confounds = True
+            signal_c_hvc = estimator.fit_transform(
+                [input_img, input_img], confounds=confounds
+            )
+
+            for s1, s2 in zip(signal_c, signal_c_hvc):
```

Comment:
why do we need to do a zip here ?

Reply:
because the transform() of multimaskers return list of arrays so we can to compare them in a pairwise fashion